File: aiko-backend-api/src/main/java/com/aiko/aikobackendapi/domain/Parada.java
package com.aiko.aikobackendapi.domain;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@ToString
@EqualsAndHashCode
public class Parada implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String name;
    private double latitude;
    private double longitude;

    @JsonBackReference
    @ManyToMany(mappedBy = "paradas")
    private List<Linha> linhas = new ArrayList();

    public Parada(String name, double latitude, double longitude) {
        super();
        this.name = name;
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public Parada() {
        super();
    }

}

File: teste-back-end/src/entity/Parada.ts
import {
  Entity, PrimaryGeneratedColumn, Column, ManyToMany,
} from 'typeorm';
import Linha from './Linha';

@Entity()
export default class Parada {
  constructor(name: string, latitude: number, longitude: number) {
    this.name = name;
    this.latitude = latitude;
    this.longitude = longitude;
  }

  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @Column({ type: 'double precision' })
  latitude: number;

  @Column({ type: 'double precision' })
  longitude: number;

  @ManyToMany(() => Linha, (linha) => linha.paradas)
  linhas: Linha[];
}
